core/src/main/java/io/cloudevents/core/message/StructuredMessageReader.java

```diff
@@ -34,11 +34,15 @@
 public interface StructuredMessageReader {
 
     /**
-     * @param visitor
+     * Read self using the provided writer.
+     *
+     * @param <R>    the return type of the {@link StructuredMessageWriter}
+     * @param writer the writer to use to write out the message
+     * @return the return value returned by {@link StructuredMessageWriter#setEvent(EventFormat, byte[])}
      * @throws CloudEventRWException If something went wrong when
      * @throws IllegalStateException If the message is not a valid structured message
      */
-    <T> T read(StructuredMessageWriter<T> visitor) throws CloudEventRWException, IllegalStateException;
+    <R> R read(StructuredMessageWriter<R> writer) throws CloudEventRWException, IllegalStateException;
 
     default CloudEvent toEvent() throws CloudEventRWException, IllegalStateException {
```

Comment:
I see these two methods are documented in extending interface. Why?

Reply:
I mean the `toEvent()` methods.

Reply:
I expect the user to look in this interface for this method, not in `StructuredMessageReader`. Does that makes sense?